### PR TITLE
adding go.mod support (removing vendor) and replacing encoding/json with json-iterator package.

### PR DESCRIPTION
**Pull request by pawels-optimizely**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,6 @@
+env:
+  - GO111MODULE=on
+
 language: go
 go:
   - 1.12.x
```
Comment (mikeng13) [inline .travis.yml]: I think we can consider adding 1.13.x here

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module github.com/optimizely/go-sdk
+
+go 1.12
```
Comment (mikeng13) [inline go.mod]: Should this be 13?

Comment (pawels-optimizely) [reply go.mod]: I think we should stay with 1.12 for now, 1.13 has been just released. We should wait and see if there are no major problems reported by go community.

File: .travis.yml (hunk 1 above)
Comment (pawels-optimizely) [reply .travis.yml]: I think we should stay with 1.12 for now, 1.13 has been just released. We should wait and see if there are no major problems reported by go community.

**Pull request merged**